\subsection{PHAROS}
\label{subsec:pharos}



 
The \href{https://druggablegenome.net/}{Illuminating the Druggable Genome} was a programme initiated by the National Institutes of Health (NIH) in 2014 and today is a worldwide collaboration between the University of New Mexico, Icahn School of Medicine, Mount Sinai, EMBL-EBI, the Novo Nordisk Foundation Center for Protein Research (University of Copenhagen), and the University of Miami \cite{pharos2016}.  Created in 2014, IDG was initially motivated to shed light to 1700 targets from four privileged drug target families \cite{santos2016}: G-Protein Coupled Receptors (GPCRs), kinases, ion channels, and nuclear receptors \cite{pharos2016}. Nevertheless, IDG is moving beyond those 4 families and now considers all 20k human coding genes \cite{pharos2016} based on phylogenecity, function, target development level, disease association, tissue expression, chemical ligand and substrate characteristics, and target-family specific characteristics. They developed the \href{http://drugtargetontology.org}{Drug Target Ontology (DTO)} \cite{lin2017}, also available on \href{http://github.com/DrugTargetOntology/DTO}{GitHub} and the \href{http://bioportal.bioontology.org/ontologies/DTO}{NCBO Bioportal}.

\noindent IDG has two distinct resources:

\begin{itemize}
    \item \href{http://juniper.health.unm.edu/tcrd/}{Target Central Resource Database (TCRD)} is the central resource behind the Illuminating the Druggable Genome Knowledge Management Center (IDG-KMC)
    
    \item \href{https://pharos.nih.gov/idg/index}{PHAROS}, as the library of Alexandria ($\phi \alpha \rho o \zeta$), is the user interface that presents TCRD information
    
\end{itemize}

\subsubsection{Data}
\begin{itemize}

    \item TCRD can be downloaded \href{http://juniper.health.unm.edu/tcrd/download/}{here}
    
    \item Information about its content can be found \href{http://habanero.health.unm.edu/tcrd/old/content.html}{here}
    
    \item A list of the datasources used can be found \href{http://targetcentral.ws/Pharos}{here}
    
\end{itemize}

Among PHAROS resources is \href{http://amp.pharm.mssm.edu/Harmonizome/about}{Harmonizome}, that aims to integrate a wide collection of public, disjoint datasets from multiple, internationally recognised datasets (n=114 in October 2018) from 66 different databases that gather information about genomics, epigenetics, transcriptomics, metabolomics, cell lines, diseases, physical interactions, drugs, and curated biomedical literature about mammalian cells \cite{harmonizome2016}. Actualised statistics of Harmonizome can be found \href{http://amp.pharm.mssm.edu/Harmonizome/about}{elsewhere}.

\textbf{Problem}: Harmonizome does not contain the raw values, just values in the set ${1,-1}$. This is a big problem. What about all the information contained in p-values? GDAs with p-values are just ones or zeroes in Harmonizome.

\subsubsection{Scores}

\begin{itemize}
    \item \texttt{novelity\_score} as measure of extent to which the published literature refers to the target. What is the formula?
    \item \texttt{pubmed\_score} described \href{https://pharos.nih.gov/idg/pmscore}{elsewhere}
    \item \texttt{data\_availability\_score} as an indication of the current information coverage at \href{http://amp.pharm.mssm.edu/Harmonizome/about}{Harmonizome}
\end{itemize}

It is important to realize that target ranking based on individual parameters is only a first step in target prioritization. While there are examples of target prioritization using individual parameters such as GO or DO terms (21), in general, target prioritization is heavily contextual, where the context could be a disease state or a biological process.

It is possible to make comparisons between targets: \url{https://pharos.nih.gov/idg/targets/compare?q=Q05586,Q9UBN1}

\subsubsection{Limitations}
\begin{itemize}
    \item Loss of information because of the binary nature of the files: e.g. the uncertainty score of the p-value is reduced to a mere binary variable with information related to "was this gene significant?" according to an arbitrary threshold.
    
    \item 
\end{itemize}